src/monero_ux_stax.c

```diff
@@ -0,0 +1,453 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *****************************************************************************/
+
+#if defined(UI_STAX)
+
+#include "os.h"
+#include "ux.h"
+#include "cx.h"
+#include "monero_types.h"
+#include "monero_api.h"
+#include "monero_vars.h"
+
+#include "monero_ux_msg.h"
+#include "os_io_seproxyhal.h"
+#include "string.h"
+#include "glyphs.h"
+
+#include "nbgl_use_case.h"
+
+#define QUIT_TOKEN     0
+#define CONTINUE_TOKEN 1
+/* ----------------------------------------------------------------------- */
+/* ---                         Stax  UI layout                         --- */
+/* ----------------------------------------------------------------------- */
+
+typedef struct {
+    nbgl_layoutTagValue_t tagValuePair[6];
+    nbgl_layoutTagValueList_t tagValueList;
+    nbgl_pageInfoLongPress_t infoLongPress;
+    char timelockBuffer[23];
+    char feeBuffer[23];
+    char changeBuffer[23];
+    nbgl_pageContent_t content;
+    nbgl_pageNavigationInfo_t nav;
+} TransactionContext_t;
+
+static TransactionContext_t transactionContext;
+static nbgl_page_t* pageContext;
+
+static void release_context(void) {
+    if (pageContext != NULL) {
+        nbgl_pageRelease(pageContext);
+        pageContext = NULL;
+    }
+}
+
+/* -------------------------------------- LOCK--------------------------------------- */
+
+void ui_menu_pinlock_display() {
+    struct {
+        bolos_ux_t ux_id;
+        // length of parameters in the u union to be copied during the syscall
+        unsigned int len;
+        union {
+            struct {
+                unsigned int cancellable;
+            } validate_pin;
+        } u;
+
+    } ux_params;
+
+    os_global_pin_invalidate();
+    G_monero_vstate.protocol_barrier = PROTOCOL_LOCKED_UNLOCKABLE;
+    ux_params.ux_id = BOLOS_UX_VALIDATE_PIN;
+    ux_params.len = sizeof(ux_params.u.validate_pin);
+    ux_params.u.validate_pin.cancellable = 0;
+    os_ux((bolos_ux_params_t*)&ux_params);
+    ui_menu_main_display();
+}
+
+/* -------------------------------- INFO UX --------------------------------- */
+
+static void ui_menu_info_action(void) {
+    if (G_monero_vstate.protocol_barrier == PROTOCOL_LOCKED) {
+        ui_menu_pinlock_display();
+    } else {
+        ui_menu_main_display();
+    }
+}
+
+void ui_menu_info_display2(unsigned int value __attribute__((unused)),
+                           const char* line1 __attribute__((unused)),
+                           const char* line2 __attribute__((unused))) {
+    nbgl_useCaseStatus("Transaction\ncancelled", false, ui_menu_info_action);
+}
+
+void ui_menu_info_display(unsigned int value __attribute__((unused))) {
+    nbgl_useCaseStatus("Security Error", false, ui_menu_info_action);
```

Comment:
Same here, you could have a `ui_menu_show_security_error()` which redirect where needed.